addons/mail/static/src/model/model_field.js

```diff
@@ -249,7 +249,11 @@ export class ModelField {
             return newVal;
         } else if (this.fieldType === 'relation') {
             // Deprecated. Used only to support old syntax: `[...[name, value]]` command
-            return newVal.map(([name, value]) => new FieldCommand(name, value));
+            if (newVal[0] && ['clear', 'insert', 'insert-and-replace', 'insert-and-unlink', 'link', 'replace', 'unlink', 'unlink-all'].includes(newVal[0][0])) {
```

Comment:
should check that `newVal` is in array before accessing `[0]` (it could also be a record, and if the record had the `0` key/field I guess the check would not work as expected)